apps/emqx_bridge/src/emqx_action_info.erl

```diff
@@ -0,0 +1,197 @@
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+%% @doc The module which knows everything about actions.
+
+%% NOTE: it does not cover the V1 bridges.
+
+-module(emqx_action_info).
+
+-export([
+    action_type_to_connector_type/1,
+    action_type_to_bridge_v1_type/1,
+    bridge_v1_type_to_action_type/1,
+    is_action_type/1,
+    registered_schema_modules/0
+]).
+
+-callback bridge_v1_type_name() -> atom().
+-callback action_type_name() -> atom().
+-callback connector_type_name() -> atom().
+-callback schema_module() -> atom().
+
+-optional_callbacks([bridge_v1_type_name/0]).
+
+%% ====================================================================
+%% Hadcoded list of info modules for actions
+%% TODO: Remove this list once we have made sure that all relevants
+%% apps are loaded before this module is called.
+%% ====================================================================
+
+-if(?EMQX_RELEASE_EDITION == ee).
+hard_coded_action_info_modules_ee() ->
+    [
+        emqx_bridge_kafka_action_info,
+        emqx_bridge_azure_event_hub_action_info
+    ].
+-else.
+hard_coded_action_info_modules_ee() ->
+    [].
+-endif.
+
+hard_coded_action_info_modules_common() ->
+    [].
+
+hard_coded_action_info_modules() ->
+    hard_coded_action_info_modules_common() ++ hard_coded_action_info_modules_ee().
+
+%% ====================================================================
+%% API
+%% ====================================================================
+
+action_type_to_connector_type(Type) when not is_atom(Type) ->
```

Comment:
Q: there are some places that already assume the convention of having `connector_name = action_name` (such as some config checks/callbacks).  If this callback seems to allow different names, wouldn't that lead to future problems?

Reply:
I think we should fix those places that relay on this assumption that is true today but might not be true in the future. I think most places already do the conversion?

Reply:
Well, we had agreed on that convention, so I'm not sure _those_ places should be changed.  Unless we retracted that convention?

Reply:
Ok, sorry. I know that we decided to use this convention for the existing bridges but not that we decided that this would be true for all future bridges as well. We can very easily enforce this convention by removing the conversion function and change the places that is calling it but I think this can be done in a different PR?

Reply:
Agreed, this doesn't need to be done here.